mmcdata.py

```diff
@@ -84,16 +86,20 @@ def _read_ascii(self,f):
         self.dataSetLength = 0
         data = []
         while True:
-            line = f.readline()
-            if line == '':
-                break
+            #line = f.readline()
+            #if line == '':
+            #    break
             recordheader = read_ascii_recordheader(f);
-            recordarray = read_ascii_records(f,self.description['levels'])
-            data.append([recordheader, recordarray])
-            self.dataSetLength += 1
+            if(not bool(recordheader)):
```

Comment:
It is not clear what `read_ascii_recordheader` returns. I think you can initialize the recordheader to {} and return an empty dict or None; then check `if recordheader is None` or `if len(recordheader) == 0`. As written, the boolean cast is puzzling, and the behavior is ambiguous. For example, is the behavior of `bool(a_dict)` generally well defined?

The other way to write this section is to just assume that the recordheader is valid and wrap your code around a try/except/else structure.

Reply:
Ok, I'll go with your more explicit test condition suggestion of 
if len(recordheader)==0 or recordheader is None: